Source: Hoshpak
Upstream: no
Reason: mode infix was removed from the two functions in linux-4.19
--- kernel/nvidia-drm/nvidia-drm-connector.c	2018-11-06 18:13:04.524177827 +0100
+++ kernel/nvidia-drm/nvidia-drm-connector.c	2018-11-06 18:13:44.110540624 +0100
@@ -30,6 +30,8 @@
 #include "nvidia-drm-utils.h"
 #include "nvidia-drm-encoder.h"
 
+#include <linux/version.h>
+
 #include <drm/drm_crtc_helper.h>
 
 #include <drm/drm_atomic.h>
@@ -226,8 +228,13 @@
 
 
     if (nv_connector->edid != NULL) {
-        drm_mode_connector_update_edid_property(
-            connector, nv_connector->edid);
+        #if LINUX_VERSION_CODE < KERNEL_VERSION(4, 19, 0)
+            drm_mode_connector_update_edid_property(
+                connector, nv_connector->edid);
+        #else
+            drm_connector_update_edid_property(
+                connector, nv_connector->edid);
+        #endif
     }
 
     while (1) {
--- kernel/nvidia-drm/nvidia-drm-encoder.c	2018-11-06 18:12:59.580066311 +0100
+++ kernel/nvidia-drm/nvidia-drm-encoder.c	2018-11-06 18:13:28.428920260 +0100
@@ -30,6 +30,8 @@
 #include "nvidia-drm-connector.h"
 #include "nvidia-drm-crtc.h"
 
+#include <linux/version.h>
+
 #include <drm/drm_crtc_helper.h>
 
 #include <drm/drm_atomic.h>
@@ -216,7 +218,11 @@
 
     /* Attach encoder and connector */
 
-    ret = drm_mode_connector_attach_encoder(connector, encoder);
+    #if LINUX_VERSION_CODE < KERNEL_VERSION(4, 19, 0)
+        ret = drm_mode_connector_attach_encoder(connector, encoder);
+    #else
+        ret = drm_connector_attach_encoder(connector, encoder);
+    #endif
 
     if (ret != 0) {
         NV_DRM_DEV_LOG_ERR(
